BeardedSpice/Tabs/VLCTabAdapter.m

```diff
@@ -0,0 +1,141 @@
+//
+//  VLCTabAdapter.m
+//  BeardedSpice
+//
+//  Created by Max Borghino on 2106-03-06
+//  Copyright (c) 2015 Tyler Rhodes / Jose Falcon. All rights reserved.
+//
+
+#import "VLCTabAdapter.h"
+#import "VLC.h"
+#import "runningSBApplication.h"
+#import "NSString+Utils.h"
+#import "MediaStrategy.h"
+
+#define APPNAME         @"VLC"
+#define APPID           @"org.videolan.vlc"
+
+@implementation VLCTabAdapter
+
++ (NSString *)displayName{
+
+    return APPNAME;
+}
+
++ (NSString *)bundleId{
+
+    return APPID;
+}
+
+- (NSString *)title {
+
+    @autoreleasepool {
+        VLCApplication *vlc = (VLCApplication *)[self.application sbApplication];
+        NSString *title;
+        if (![NSString isNullOrEmpty:vlc.nameOfCurrentItem])
+            title = vlc.nameOfCurrentItem;
+
+        if ([NSString isNullOrEmpty:title]) {
+            title = NSLocalizedString(@"No Track", @"SpotifyTabAdapter");
+        }
+
+        return [NSString stringWithFormat:@"%@ (%@)", title, APPNAME];
+    }
+}
+- (NSString *)URL{
+
+    return @"VLC";
+}
+
+// We have only one window.
+- (NSString *)key{
+
+    return @"A:VLC";
+}
+
+// We have only one window.
+-(BOOL) isEqual:(__autoreleasing id)otherTab{
+
+    if (otherTab == nil || ![otherTab isKindOfClass:[VLCTabAdapter class]]) return NO;
+
+    return YES;
+}
+
+// try to find the document which will respond to controls
+// NOTE: this feels brittle, it seems there should be a better way
+- (VLCDocument*) activeDocument {
+    VLCDocument *doc = NULL;
+    VLCApplication *vlc = (VLCApplication *)[self.application sbApplication];
+    if (vlc) {
+        NSEnumerator *e = [[vlc windows] objectEnumerator];
+        id o;
+        while ((o = [e nextObject]) && doc == NULL) {
+            NSLog(@"VLC window %@: %lu: %@", o, [o index], [o name]);
+            if ([[o name] isEqualToString: [vlc nameOfCurrentItem]]) {
```

Comment:
Yeah I thought about that, and it seems like it might be better (less likely to conflict with anything else), but the window doesn't have a path, and I can't seem to get the document to show me the path either. I suspect this is pilot error.

The  current scheme (app's name of current item == window's name) seems to work ok.